### PR TITLE
Enable sonarcloud scanning and coverage

### PR DESCRIPTION
**Pull request by micbar**
## Description

Enable Sonarcloud Scanning

@phil-davis It shows ~18% oc coverage. codacy shows more than 50% ?

I cannot figure that out.

### CODE REVIEW

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/ocis/425/13/6
```
INFO: Sensor Go Cover sensor for Go coverage [go]
INFO: Load coverage report from '/drone/src/coverage/store_coverage.out'
ERROR: Error parsing coverage info for file /drone/src/coverage/store_coverage.out: Invalid go coverage, expect 'mode:' on the first line.
INFO: Load coverage report from '/drone/src/coverage/ocis-phoenix_coverage.out'
ERROR: Error parsing coverage info for file /drone/src/coverage/ocis-phoenix_coverage.out: Invalid go coverage, expect 'mode:' on the first line.
INFO: Load coverage report from '/drone/src/coverage/accounts_coverage.out'
INFO: Load coverage report from '/drone/src/coverage/ocs_coverage.out'
INFO: Load coverage report from '/drone/src/coverage/storage_coverage.out'
INFO: Load coverage report from '/drone/src/coverage/settings_coverage.out'
INFO: Load coverage report from '/drone/src/coverage/proxy_coverage.out'
INFO: Load coverage report from '/drone/src/coverage/thumbnails_coverage.out'
INFO: Load coverage report from '/drone/src/coverage/ocis-pkg_coverage.out'
INFO: Load coverage report from '/drone/src/coverage/glauth_coverage.out'
ERROR: Error parsing coverage info for file /drone/src/coverage/glauth_coverage.out: Invalid go coverage, expect 'mode:' on the first line.
INFO: Load coverage report from '/drone/src/coverage/webdav_coverage.out'
ERROR: Error parsing coverage info for file /drone/src/coverage/webdav_coverage.out: Invalid go coverage, expect 'mode:' on the first line.
INFO: Load coverage report from '/drone/src/coverage/konnectd_coverage.out'
ERROR: Error parsing coverage info for file /drone/src/coverage/konnectd_coverage.out: Invalid go coverage, expect 'mode:' on the first line.
WARN: File 'github.com/owncloud/ocis/accounts/pkg/assets/embed.go' is not included in the project, ignoring coverage
INFO: Sensor Go Cover sensor for Go coverage [go] (done) | time=332ms
```

I think the messages about `Invalid go coverage, expect 'mode:' on the first line` are because those extensions do not have any tests, and so the `*_coverage.out` is a completely empty file.

The overall coverage etc still seems to get reported - so it is not a fatal error.

File: .drone.star [hunk 1]
```diff
@@ -274,6 +274,14 @@ def uploadCoverage(ctx):
           'mc mirror cache/cache/%s/%s/coverage coverage/' % (ctx.repo.slug, ctx.build.commit + '-${DRONE_BUILD_NUMBER}'),
         ]
       },
+      {
+        'name': 'show-before',
+        'image': 'owncloud/alpine:latest',
+        'pull': 'always',
+        'commands': [
+          'cat coverage/*',
```
Comment (phil-davis) [inline .drone.star:282]: Do you want to keep this "debugging" or or not?

File: .drone.star [hunk 2]
```diff
@@ -284,6 +292,27 @@ def uploadCoverage(ctx):
           },
         },
       },
+      {
+        'name': 'show-after',
+        'image': 'owncloud/alpine:latest',
+        'pull': 'always',
+        'commands': [
+          'cat coverage/*',
```
Comment (phil-davis) [inline .drone.star:300]: Do you want to keep this "debugging" or or not?

**Review by phil-davis — APPROVED** [review]
Looks fine and has produced SonarCloud data.

The actual coverage is quite low - because it is based on "unit" tests. Actually at the moment most of the coverage is by running the core API end-to-end test suite, and we don't have a way to collect coverage stats while that runs.

File: .drone.star (hunk 2 above)
Comment (micbar) [reply .drone.star]: removed

File: .drone.star (hunk 1 above)
Comment (micbar) [reply .drone.star]: removed

**Comment by phil-davis** [comment]
Data for `owncloud/ocis` master is now available at https://sonarcloud.io/dashboard?id=owncloud_ocis

**Pull request merged**